Apply mix format and update query test for CI matrix

fastjames opened the pull request:
In order to maintain consistent code formatting, apply the outstanding
change from `mix format`. This should fix the failing format step in CI.

fastjames commented:
I'm not sure how I missed it in my last PR, but here's the the fix for the failing `mix format` step in CI.

fastjames commented:
I see what's going on with the failing test:

```
  1) test queryable add coalesces in WHERE == clause when custom coalesce function returns nil for multiple sort_fields (QuartoTest)
     test/quarto_test.exs:657
     Assertion with == failed
     code:  assert "#Ecto.Query<from p0 in Quarto.Post, where: (coalesce(p0.position, ^100) == ^3 and (coalesce(p0.id, ^10) > ^10 and ^true)) or\n  (coalesce(p0.position, ^100) < ^3 and ^true and ^true), order_by: [desc: coalesce(p0.position, ^100), asc: coalesce(p0.id, ^10)], limit: ^6>\n" == inspect(queryable) <> "\n"
     left:  "#Ecto.Query<from p0 in Quarto.Post, where: (coalesce(p0.position, ^100) == ^3 and (coalesce(p0.id, ^10) > ^10 and ^true)) or\n  (coalesce(p0.position, ^100) < ^3 and ^true and ^true), order_by: [desc: coalesce(p0.position, ^100), asc: coalesce(p0.id, ^10)], limit: ^6>\n"
     right: "#Ecto.Query<from p0 in Quarto.Post, where: coalesce(p0.position, ^100) == ^3 and (coalesce(p0.id, ^10) > ^10 and ^true) or coalesce(p0.position, ^100) < ^3 and ^true and ^true, order_by: [desc: coalesce(p0.position, ^100), asc: coalesce(p0.id, ^10)], limit: ^6>\n"
     stacktrace:
       test/quarto_test.exs:674: (test)
```

The test compares the output of IO.inspect/2 with the expected string. It looks like IO.inspect/2 is somehow being told to pretty-print the output, which causes it to drop in a `\n` at char position 80 which invalidates the string match.

maartenvanvliet commented:
🙏 thx

fastjames commented:
Well, this is annoying -- the output of `inspect` appears to shift slightly across elixir versions.

elixir 1.11.4 / OTP 23:

```
     left:  "#Ecto.Query<from p0 in Quarto.Post, where: (coalesce(p0.position, ^100) == ^3 and (coalesce(p0.id, ^10) > ^10 and ^true)) or\n  (coalesce(p0.position, ^100) < ^3 and ^true and ^true), order_by: [desc: coalesce(p0.position, ^100), asc: coalesce(p0.id, ^10)], limit: ^6>"
     right: "#Ecto.Query<from p0 in Quarto.Post, where: coalesce(p0.position, ^100) == ^3 and (coalesce(p0.id, ^10) > ^10 and ^true) or coalesce(p0.position, ^100) < ^3 and ^true and ^true, order_by: [desc: coalesce(p0.position, ^100), asc: coalesce(p0.id, ^10)], limit: ^6>"
```

elixir 1.12.3 / OTP 24:

```
     left:  "#Ecto.Query<from p0 in Quarto.Post, where: (coalesce(p0.position, ^100) == ^3 and (coalesce(p0.id, ^10) > ^10 and ^true)) or\n  (coalesce(p0.position, ^100) < ^3 and ^true and ^true), order_by: [desc: coalesce(p0.position, ^100), asc: coalesce(p0.id, ^10)], limit: ^6>"
     right: "#Ecto.Query<from p0 in Quarto.Post, where: coalesce(p0.position, ^100) == ^3 and (coalesce(p0.id, ^10) > ^10 and ^true) or coalesce(p0.position, ^100) < ^3 and ^true and ^true, order_by: [desc: coalesce(p0.position, ^100), asc: coalesce(p0.id, ^10)], limit: ^6>"

elixir 1.13.4 / OTP 25:

```
     left:  "#Ecto.Query<from p0 in Quarto.Post, where: (coalesce(p0.position, ^100) == ^3 and (coalesce(p0.id, ^10) > ^10 and ^true)) or\n  (coalesce(p0.position, ^100) < ^3 and ^true and ^true), order_by: [desc: coalesce(p0.position, ^100), asc: coalesce(p0.id, ^10)], limit: ^6>foo"
     right: "#Ecto.Query<from p0 in Quarto.Post, where: (coalesce(p0.position, ^100) == ^3 and (coalesce(p0.id, ^10) > ^10 and ^true)) or\n  (coalesce(p0.position, ^100) < ^3 and ^true and ^true), order_by: [desc: coalesce(p0.position, ^100), asc: coalesce(p0.id, ^10)], limit: ^6>"
```

We can strip out the newlines, but it looks like there are some extra parentheses in elixir 1.13 that don't appear in the earlier outputs. I will figure out a way around that.

fastjames commented:
I believe I have found a way around the `inspect/2` changes. It is not elegant but it does work as advertised. A future improvement might involve replacing the string comparison with tests against a map representation of the query (by passing `structs: false` to inspect). For now, this seems less disruptive.

maartenvanvliet commented:
The format check can be done on 1.13 alone in the test matrix. Couple of my other repo's show how that can be achieved. Let me know if you need help.

fastjames commented:
OK, I updated the test with a tag and changed the workflow to skip that tagged test on older elixir versions in the matrix. Does that look like what you expected to see?

maartenvanvliet commented:
Looks good!
